db/silkworm/etl/file_provider.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <silkworm/etl/file_provider.hpp>
+
+namespace silkworm::etl{
+
+FileProvider::FileProvider(Buffer * buffer, int id) {
+    filename_ = "tmp" + std::to_string(id);
+    file_.open(filename_, std::ios_base::in | std::ios_base::out | std::ios_base::trunc);
+    auto entries{buffer->get_entries()};
+    for(auto entry: entries) {
+        auto writes{std::string()};
+        unsigned char length_key[4];
+        unsigned char length_value[4];
+        boost::endian::store_big_u32(length_key, entry.key.size());
+        boost::endian::store_big_u32(length_value, entry.value.size());
+        writes.append(std::string((const char *) length_key, 4));
+        writes.append(std::string((const char *) length_value, 4));
+        writes.append(std::string((const char *) entry.key.data(), entry.key.size()));
+        writes.append(std::string((const char *) entry.value.data(), entry.value.size()));
+        file_ << writes;
+    }
+    file_.seekp(0);
```

Comment:
For file stream it doesn't really matter but I'd prefer to use seekg instead of seekp to clarify you're setting the pointer for subsequent reads.